index.js

```diff
@@ -616,14 +616,25 @@ RedisClient.prototype.connection_gone = function (why, error) {
             if (this.retry_delay instanceof Error) {
                 error = this.retry_delay;
             }
+            // Handle object pattern
+            if (typeof this.retry_delay === 'object' && !(this.retry_delay instanceof Error) && this.retry_delay !== null) {
```

Comment:
The `&& !(this.retry_delay instanceof Error)` is a redundant check. Please just use a `else if` instead.